packages/components/src/table/index.js

```diff
@@ -73,6 +73,13 @@ class TableCard extends Component {
 				/* eslint-enable react/no-did-update-set-state */
 			}
 		}
+		if ( ! isLoading && prevIsLoading && ! isEqual( headers, prevHeaders ) ) {
+			/* eslint-disable react/no-did-update-set-state */
+			this.setState( {
+				showCols: headers.map( ( { key, hiddenByDefault } ) => ! hiddenByDefault && key ).filter( Boolean ),
```

Comment:
Why not use an update function for `setState`? It would guarantee `headers` is up-to-date when the state is set: https://reactjs.org/docs/react-component.html#setstate

Reply:
Will there be any difference using a function or an object in `setState` considering `headers` is a prop? I think `props` will be up-to-date in both cases, so there is no need to use a function for `setState`. Am I missing something?

Reply:
Yes, that's true. I must have assumed headers was in state. I don't think `setState` is susceptible to props changing in between, and if it were, another update would soon follow.